pymeasure/instruments/keysight/keysightDSOX1102G.py

```diff
@@ -0,0 +1,647 @@
+                       scale: float = None):
+        """ Set up timebase. Unspecified parameters are not modified. Modifying a single parameter might
+        impact other parameters. Refer to oscilloscope documentation and make multiple consecutive calls
+        to channel_setup if needed.
+
+        :param mode: Timebase mode, can be "main", "window", "xy", or "roll".
+        :param offset: Offset in seconds between trigger and center of screen.
+        :param horizontal_range: Full-scale range in seconds.
+        :param scale: Units-per-division in seconds."""
+
+        if mode is not None: self.timebase_mode = mode
+        if offset is not None: self.timebase_offset = offset
+        if horizontal_range is not None: self.timebase_range = horizontal_range
+        if scale is not None: self.timebase_scale = scale
+
+    def download_image(self, format_: str = "png", color_palette: str = "color"):
+        """ Get image of oscilloscope screen in bytearray of specified file format.
+
+        :param format_: "bmp", "bmp8bit", or "png"
+        :param color_palette: "color" or "grayscale"
+        """
+        query = f":DISPlay:DATA? {format_}, {color_palette}"
+        # Using binary_values query because default interface does not support binary transfer
+        img = self.adapter.connection.query_binary_values(query, datatype="s")
+        return bytearray(img[0])
+
+    def download_data(self, source: str, points: int = 62500):
+        """ Get data from specified source of oscilloscope. Returned objects are a np.ndarray of data
+        values (no temporal axis) and a dict of the waveform preamble, which can be used to build the
+        corresponding time values for all data points.
+
+        Multimeter will be stopped for proper acquisition.
+
+        :param source: measurement source, can be "channel1", "channel2", "function", "fft", "wmemory1",
+                        "wmemory2", or "ext".
+        :param points: nb of points to acquire. Note that oscilloscope may return less points than
+                        specified, this is not an issue of this library. Can be 100, 250, 500, 1000,
+                        2000, 5000, 10000, 20000, 50000, or 62500.
+
+        :return data_ndarray, waveform_preamble_dict: see waveform_preamble property for dict format.
+        """
+        # TODO: Consider downloading from multiple sources at the same time.
+        self.waveform_source = source
+        self.waveform_points_mode = "normal"
+        self.waveform_points = points
+
+        preamble = self.waveform_preamble
+        data_bytes = self.waveform_data
+        return np.array(data_bytes), preamble
+
+    def _ch(self, chNb):
+        """
+        Reads setup data from channel and converts it to a more convenient dict of values.
+        """
+        if chNb == 1:
+            ch_prefix = ":channel1"
+        elif chNb == 2:
+            ch_prefix = ":channel2"
+        else:
+            raise ValueError("Bad channel nb.")
```

Comment:
Please avoid unnecessarily terse/abbreviated formulations. Same in general for variables like `chNb` above - in Python in my experience, `channel_number` would be totally ok and even preferred because of increased clarity. :-)